src/Language/Mimsa/Typechecker/Unify.hs

```diff
@@ -90,8 +97,12 @@ unifyStrict tyA tyB =
         else unify tyA tyB
     (a, b) -> unify a b
 
-getTypeOrFresh :: Annotation -> Name -> Map Name MonoType -> TcMonad MonoType
-getTypeOrFresh ann name map' =
+getRecordItemType ::
+  Annotation ->
+  Name ->
+  Map Name MonoType ->
+  TcMonad MonoType
+getRecordItemType ann name map' =
   case M.lookup name map' of
     Just found -> pure found
-    _ -> getUnknown ann
+    _ -> throwError (MissingRecordTypeMember ann name map')
```

Comment:
Because we walk the syntax tree before inference to work out the shapes of all records, we no longer need to fill them in as we go.